app/services/billable_metrics/destroy_service.rb

```diff
@@ -2,24 +2,36 @@
 
 module BillableMetrics
   class DestroyService < BaseService
-    def destroy(id)
-      metric = result.user.billable_metrics.find_by(id:)
-      return result.not_found_failure!(resource: 'billable_metric') unless metric
-
-      metric.destroy!
+    def self.call(...)
+      new(...).call
+    end
 
-      result.billable_metric = metric
-      result
+    def initialize(metric:)
+      @metric = metric
+      super
     end
 
-    def destroy_from_api(organization:, code:)
-      metric = organization.billable_metrics.find_by(code:)
+    def call
       return result.not_found_failure!(resource: 'billable_metric') unless metric
 
-      metric.destroy!
+      ActiveRecord::Base.transaction do
+        metric.discard
```

Comment:
Should we use `discard!` to ensure the action is executed?

Reply:
good idea!